portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,21 +31,44 @@
 import javax.servlet.http.HttpServletResponse;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+@WebServlet("/comments")
 public class DataServlet extends HttpServlet {
-
-    ArrayList<String> messages = new ArrayList<>(Arrays.asList("Hello", "Hola", "Bonjour"));
-
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        ArrayList<String> messages = new ArrayList<>();
+        for(Entity entity : results.asIterable()){
+            messages.add((String)entity.getProperty("comment"));
```

Comment:
All of these hardcoded string literals which are tied to your data model should preferably be pulled out into static constants (what happens now if you typo them in one place but not the other?). If you need them between classes, you can pull those constants into their own class.

Reply:
Added a Constants.java file with all of the hardcoded strings